x-pack/platform/plugins/shared/ml/public/application/model_management/deployment_params_mapper.ts

```diff
@@ -152,8 +152,13 @@ export class DeploymentParamsMapper {
       number_of_allocations: maxValue,
       min_number_of_allocations:
         Math.floor(levelValues.min / threadsPerAllocation) ||
-        // in any env, allow scale down to 0 only for "low" vCPU usage
-        (params.vCPUUsage === 'low' ? this.minAllowedNumberOfAllocation : 1),
+        // For serverless env, always allow scale down to 0
+        // For other envs, allow scale down to 0 only for "low" vCPU usage
+        (this.showNodeInfo === false
+          ? this.minAllowedNumberOfAllocation
+          : params.vCPUUsage === 'low'
+          ? this.minAllowedNumberOfAllocation
+          : 1),
```

Comment:
could you please add unit tests for it?

Reply:
Also, since the condition has become more complex and now includes a nested ternary operator, we might want to consider improving readability.: 
```suggestion
let minAllocations = Math.floor(levelValues.min / threadsPerAllocation);

if (!minAllocations) {
  if (!this.showNodeInfo || params.vCPUUsage === 'low') {
    minAllocations = this.minAllowedNumberOfAllocation;
  } else {
    minAllocations = 1;
  }
}
```